tests/test_integrations.py

```diff
@@ -108,3 +110,74 @@ def test_script(case):
     assert exp_out == out
     assert exp_rtn == p.returncode
 
+
+@skip_if_on_windows
+@pytest.mark.parametrize('cmd, fmt, exp', [
+    ('pwd', None, XONSH_PREFIX + '\n'),
+    ('echo WORKING', None, 'WORKING\n'),
+    ('ls -f', lambda out: out.splitlines().sort(), os.listdir().sort()),
+    ])
+def test_single_command(cmd, fmt, exp):
+    """The ``fmt`` parameter is a function
+    that formats the output of cmd, can be None.
+    """
+    env = dict(os.environ)
+    env['PATH'] = PATH
+    env['XONSH_DEBUG'] = '1'
+    env['XONSH_SHOW_TRACEBACK'] = '1'
+    env['RAISE_SUBPROC_ERROR'] = '1'
+    env['PROMPT'] = ''
+    xonsh = 'xonsh.bat' if ON_WINDOWS else 'xon.sh'
+    xonsh = shutil.which(xonsh, path=PATH)
+    p = subprocess.Popen([xonsh, '--no-rc'],
+                         env=env,
+                         stdin=subprocess.PIPE,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT,
+                         universal_newlines=True,
+                         )
+    try:
+        out, err = p.communicate(input=cmd, timeout=10)
+    except subprocess.TimeoutExpired:
+        p.kill()
+        raise
+    if callable(fmt):
+        out = fmt(out)
+    assert out == exp
+    assert p.returncode == 0
```

Comment:
Is there anyway that we could generealize this function so that it is essentially not copied three times in the file?  Maybe make one `check_command()` or something?